src/datasources/transaction-api/transaction-api.service.ts

```diff
@@ -860,4 +862,17 @@ export class TransactionApi implements ITransactionApi {
     });
     await this.cacheService.deleteByKey(key);
   }
+
+  private mapError(error: unknown): unknown {
+    if (error instanceof NetworkResponseError) {
+      // Map Django error as "standard" error message if present
+      const djangoError = get(error.data, 'nonFieldErrors[0]');
+      if (djangoError) {
+        return new NetworkResponseError(error.url, error.response, {
+          message: djangoError,
+        });
```

Comment:
Nit: should we abstract from the technology used by the data source? I mean, what about changing the `django` references here and in the tests for another generic concept like `transactionService`?

Reply:
And also other nits: 
- `nonFieldErrors[0]` might be extracted to a class constant, wdyt? 
- This assumes we are only interested in the first error in the payload, aren't we? I'm ok with this as long as this is ok for the clients.

Reply:
I generalised all references to "Django" (preferring "Transaction Service") in 931505d2b0fdb5eae083f4fd891acea1b14fa97b, as well as moving the path to a constant. I also added a comment explaining the reasoning for returning the first error element - a string is preferred.